test/metabase/csv_test.clj

```diff
@@ -127,3 +130,71 @@
+                                             uploads-database-id  nil
+                                             uploads-schema-name  "public"
+                                             uploads-table-prefix "uploaded_magic_"]
+            (is (thrown-with-msg?
+                 java.lang.Exception
+                 #"^You must set the `uploads-database-id` before uploading files\.$"
+                 (csv/load! file)))))
+        (testing "Database ID must be valid"
+          (mt/with-temporary-setting-values [uploads-enabled      true
+                                             uploads-database-id  -1
+                                             uploads-schema-name  "public"
+                                             uploads-table-prefix "uploaded_magic_"]
+            (is (thrown-with-msg?
+                 java.lang.Exception
+                 #"^The uploads database does not exist\."
+                 (csv/load! file)))))
+        (testing "Schema name must be set"
+          (mt/with-temporary-setting-values [uploads-enabled      true
+                                             uploads-database-id  db-id
+                                             uploads-schema-name  nil
+                                             uploads-table-prefix "uploaded_magic_"]
+            (is (thrown-with-msg?
+                 java.lang.Exception
+                 #"^You must set the `uploads-schema-name` before uploading files\.$"
+                 (csv/load! file)))))
+        (testing "Table prefix must be set"
+          (mt/with-temporary-setting-values [uploads-enabled      true
+                                             uploads-database-id  db-id
+                                             uploads-schema-name  "public"
+                                             uploads-table-prefix nil]
+            (is (thrown-with-msg?
+                 java.lang.Exception
+                 #"^You must set the `uploads-table-prefix` before uploading files\.$"
+                 (csv/load! file)))))
+        (testing "Happy path"
+          (mt/with-temporary-setting-values [uploads-enabled      true
+                                             uploads-database-id  db-id
+                                             uploads-schema-name  "public"
+                                             uploads-table-prefix "uploaded_magic_"]
+            (csv/load! file)
+            (let [new-table (t2/select-one Table {:where
+                                                  [:and [:= :db_id db-id]
+                                                        (when (seq existing-table-ids)
+                                                          [:not-in :id existing-table-ids])]})]
+              (is (str/starts-with? (:name new-table) "pokefans"))
+              (is (= "public" (:schema new-table)))
+              (is (= #{"id" "name"} (t2/select-fn-set :name Field :table_id (:id new-table)))))))))))
```

Comment:
I figured all of the tricky cases are handled by the tests for the component functions—the point of this `deftest` is to make sure everything is coordinated right.